File: src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/fun/TretaNewsCommand.java
package com.mrpowergamerbr.loritta.commands.vanilla.fun;

import com.google.common.collect.ImmutableMap;
import com.mrpowergamerbr.loritta.commands.CommandBase;
import com.mrpowergamerbr.loritta.commands.CommandCategory;
import com.mrpowergamerbr.loritta.commands.CommandContext;
import com.mrpowergamerbr.loritta.utils.LorittaUtils;
import com.mrpowergamerbr.loritta.utils.TretaNewsGenerator;
import net.dv8tion.jda.core.MessageBuilder;
import net.dv8tion.jda.core.OnlineStatus;
import net.dv8tion.jda.core.entities.Member;
import net.dv8tion.jda.core.entities.User;

import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.SplittableRandom;

public class TretaNewsCommand extends CommandBase {
	public static final String HIDE_DISCORD_TAGS = "esconderTagsDoDiscord";
	public static final String MENTION_USERS = "mencionarUsuarios";

	@Override
	public String getLabel() {
		return "tretanews";
	}

	public String getDescription() {
		return "VOOOOOOOCÊ ESTÁ ASSISTINDO TRETA NEWS ENTÃO VAMOS DIRETO PARA AS NOTÍCIAS";
	}

	public String getUsage() {
		return "[usuário1] [usuário2]";
	}

	public List<String> getExample() {
		return Arrays.asList("", "@Loritta @MrPowerGamerBR");
	}

	public Map<String, String> getDetailedUsage() {
		return ImmutableMap.<String, String>builder()
				.put("usuário1", "*(Opcional)* \"YouTuber\" sortudo que apareceu no Treta News")
				.put("usuário2", "*(Opcional)* \"YouTuber\" sortudo que apareceu no Treta News")
				.build();
	}

	@Override
	public CommandCategory getCategory() {
		return CommandCategory.FUN;
	}

	@Override
	public void run(CommandContext context) {
		if (!LorittaUtils.canUploadFiles(context)) { return; }
		try {
			User user1 = null;
			User user2 = null;

			if (context.getMessage().getMentionedUsers().size() >= 1) {
				user1 = context.getMessage().getMentionedUsers().get(0);
			}

			if (context.getMessage().getMentionedUsers().size() >= 2) {
				user2 = context.getMessage().getMentionedUsers().get(1);
			}

			if (user1 == null) {
				Member member1 = context.getGuild().getMembers().get(new SplittableRandom().nextInt(context.getGuild().getMembers().size()));

				while (member1.getOnlineStatus() == OnlineStatus.OFFLINE) {
					member1 = context.getGuild().getMembers().get(new SplittableRandom().nextInt(context.getGuild().getMembers().size()));
				}

				user1 = member1.getUser();
			}

			if (user2 == null) {
				Member member2 = context.getGuild().getMembers().get(new SplittableRandom().nextInt(context.getGuild().getMembers().size()));

				while (member2.getOnlineStatus() == OnlineStatus.OFFLINE) {
					member2 = context.getGuild().getMembers().get(new SplittableRandom().nextInt(context.getGuild().getMembers().size()));
				}

				user2 = member2.getUser();
			}

			BufferedImage base = TretaNewsGenerator.generate(context.getGuild(), context.getGuild().getMember(user1), context.getGuild().getMember(user2));

			ByteArrayOutputStream os = new ByteArrayOutputStream();
			ImageIO.write(base, "png", os);
			InputStream is = new ByteArrayInputStream(os.toByteArray());

			MessageBuilder builder = new MessageBuilder();
			builder.append(context.getAsMention(true) + "VOOOOOOCÊ ESTÁ ASSISTINDO TRETA NEWS E VAMOS DIRETO PARA AS NOTÍCIAAAAAAAAS!");

			if (context.getConfig().getCommandOptionsFor(this).getAsBoolean(MENTION_USERS)) {
				builder.append(" ");
				builder.append(user1);
				builder.append(" ");
				builder.append(user2);
			} else {

			}

			context.sendFile(is, "tretanews.png", builder.build());
		} catch (IOException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}
}

File: src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/images/TretaNewsCommand.kt
package com.mrpowergamerbr.loritta.commands.vanilla.images

import com.mrpowergamerbr.loritta.Loritta.Companion.RANDOM
import com.mrpowergamerbr.loritta.commands.AbstractCommand
import com.mrpowergamerbr.loritta.commands.CommandCategory
import com.mrpowergamerbr.loritta.commands.CommandContext
import com.mrpowergamerbr.loritta.utils.LoriReply
import com.mrpowergamerbr.loritta.utils.LorittaUtils
import com.mrpowergamerbr.loritta.utils.TretaNewsGenerator
import com.mrpowergamerbr.loritta.utils.locale.BaseLocale
import net.dv8tion.jda.core.OnlineStatus
import net.dv8tion.jda.core.entities.User
import java.util.*

class TretaNewsCommand : AbstractCommand("tretanews", category = CommandCategory.FUN) {
	override fun getDescription(locale: BaseLocale): String {
		return locale["TRETANEWS_DESCRIPTION"]
	}

	override fun getUsage(): String {
		return "[usuário1] [usuário2]"
	}

	override fun getExample(): List<String> {
		return Arrays.asList("", "@Loritta @MrPowerGamerBR")
	}

	override fun getDetailedUsage(): Map<String, String> {
		return mapOf("usuário1" to "*(Opcional)* \"YouTuber\" sortudo que apareceu no Treta News",
				"usuário2" to "*(Opcional)* \"YouTuber\" sortudo que apareceu no Treta News")
	}

	override fun needsToUploadFiles(): Boolean {
		return true
	}

	override fun run(context: CommandContext, locale: BaseLocale) {
		var user1: User? = context.getUserAt(0)
		var user2: User? =  context.getUserAt(1)

		if (user1 == null) {
			var member1 = context.guild.members[RANDOM.nextInt(context.guild.members.size)]

			while (member1.onlineStatus == OnlineStatus.OFFLINE) {
				member1 = context.guild.members[RANDOM.nextInt(context.guild.members.size)]
			}

			user1 = member1.user
		}

		if (user2 == null) {
			var member2 = context.guild.members[RANDOM.nextInt(context.guild.members.size)]

			while (member2.onlineStatus == OnlineStatus.OFFLINE) {
				member2 = context.guild.members[RANDOM.nextInt(context.guild.members.size)]
			}

			user2 = member2.user
		}

		val base = TretaNewsGenerator.generate(context.guild, context.guild.getMember(user1), context.guild.getMember(user2))

		context.reply(base.image, "tretanews.png",
				LoriReply(
						message = "VOOOOOOCÊ ESTÁ ASSISTINDO TRETA NEWS E VAMOS DIRETO PARA AS NOTÍCIAAAAAAAAS!",
						prefix = "<:fluffy:372454445721845761>"
				),
				LoriReply(
						message = "`${base.title}`",
						mentionUser = false
				),
				LoriReply(
						message = "\uD83D\uDCFA `${base.views}` **${context.locale["MUSICINFO_VIEWS"]}**, \uD83D\uDE0D `${base.likes}` **${context.locale["MUSICINFO_LIKES"]}**, \uD83D\uDE20 `${base.dislikes}` **${context.locale["MUSICINFO_DISLIKES"]}**",
						prefix = "\uD83D\uDCC8",
						mentionUser = false
				)
		)
	}
}